var/spack/repos/builtin/packages/spectrum-mpi/package.py

```diff
@@ -56,3 +56,22 @@ def setup_dependent_build_environment(self, env, dependent_spec):
         env.set('OMPI_F77', spack_f77)
 
         env.prepend_path('LD_LIBRARY_PATH', self.prefix.lib)
+
+    def setup_run_environment(self, env):
+        # Because MPI functions as a compiler we need to setup the compilers
```

Comment:
It looks like you could update `setup_dependent_build_environment` to use this function as you have done with other packages. Not a big deal but FYI

Reply:
You can't in this case, because in the `setup_run_environment` case you're setting the MPI compilers based on the compiler used for the MPI spec, and in the `setup_dependent_environment` case you're setting it based on the dependent spec (in case you're using a different compiler from your MPI).

I could factor out a separate function that takes the spec as an input and call that from both places, but it didn't seem warranted.